gapic/schema/wrappers.py

```diff
@@ -767,6 +767,23 @@ def http_opt(self) -> Optional[Dict[str, str]]:
         # TODO(yon-mg): enums for http verbs?
         return answer
 
+    @property
+    def path_params(self) -> Sequence[str]:
+        if self.http_opt is None:
+            return []
+        pattern = r'\{\w+\}'
+        return [x[1:-1] for x in re.findall(pattern, self.http_opt['url'])]
+
+    @property
+    def query_params(self) -> Set[str]:
+        if self.http_opt is None:
+            return set()
+        body = []
+        if 'body' in self.http_opt.keys():
+            body.append(self.http_opt['body'])
+        all_params = self.input.fields.keys()
```

Comment:
We can move the variable setups around a little to make it a little easier to read.
```python
params = set(self.path_params)
body = self.http_opt.get('body')
if body:
    params.add(body)

return set(self.input.fields) ^ params
```
Operations on dicts by default just use the keyspace, e.g. `assert 6 in {6: "six"}` and `assert set({6: "six"}) == {6}`.